pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1888,6 +1884,50 @@ func (service *SecurityPolicyService) buildRulePortsString(ports *[]v1alpha1.Sec
 	return util.GenerateTruncName(common.MaxNameLength, portsString, "", suffix, "", "")
 }
 
+func (service *SecurityPolicyService) buildRulePortNumberString(port v1alpha1.SecurityPolicyPort, portNumber int) string {
+	// Build the rule port number string name for non named port.
+	// This is a common case where the string is built from port definition. For instance,
+	// - protocol: TCP
+	//   port: 8282
+	//   endPort: 8286
+	// The built port number string is: 8282.8286
+	// - protocol: UDP
+	//   port: 3308
+	// The built port number string is: 3308
+	if port.Port.Type == intstr.String && portNumber != -1 {
```

Comment:
@wenyingd  Add more check for namedport case.
The UTs have been updated as well.

Reply:
What about 
```
if port.Port.Type == intstr.String && portNumber > 0 {
...
```

Reply:
It makes sense to me, Thanks for advice. Fixed.